source/agora/node/Runner.d

```diff
@@ -128,12 +126,13 @@ public Listeners runNode (Config config)
 
     if (config.registry.enabled)
     {
-        result.registry = new NameRegistry(config.registry, result.node);
-        router.registerRestInterface!(NameRegistryAPI)(result.registry);
+        auto reg = result.node.getRegistry();
+        assert(reg !is null);
+        router.registerRestInterface(reg);
         if (config.registry.dns.enabled)
         {
-            /* auto dnstask = */ runTask(() => runDNSServer(config.registry, result.registry));
-            dnstcplistener = listenTCP(config.registry.dns.port, (conn) => conn.runTCPDNSServer(result.registry),
+            /* auto dnstask = */ runTask(() => runDNSServer(config.registry, reg));
+            dnstcplistener = listenTCP(config.registry.dns.port, (conn) => conn.runTCPDNSServer(reg),
```

Comment:
Not part of your changes but `dns_tcp_listener` is easier to read.

Reply:
Indeed. I'm going to shuffle this a bit more soon, so I'll rename the variables in the process.